src/main/java/com/paloaltonetworks/osc/api/PANApplianceManagerApi.java

```diff
@@ -1,286 +1,288 @@
+import javax.ws.rs.client.ClientBuilder;
+
+import org.osc.sdk.manager.api.ApplianceManagerApi;
+import org.osc.sdk.manager.api.IscJobNotificationApi;
+import org.osc.sdk.manager.api.ManagerCallbackNotificationApi;
+import org.osc.sdk.manager.api.ManagerDeviceApi;
+import org.osc.sdk.manager.api.ManagerDeviceMemberApi;
+import org.osc.sdk.manager.api.ManagerDomainApi;
+import org.osc.sdk.manager.api.ManagerPolicyApi;
+import org.osc.sdk.manager.api.ManagerSecurityGroupApi;
+import org.osc.sdk.manager.api.ManagerSecurityGroupInterfaceApi;
+import org.osc.sdk.manager.api.ManagerWebSocketNotificationApi;
+import org.osc.sdk.manager.element.ApplianceManagerConnectorElement;
+import org.osc.sdk.manager.element.VirtualSystemElement;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.metatype.annotations.AttributeDefinition;
+import org.osgi.service.metatype.annotations.ObjectClassDefinition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.paloaltonetworks.panorama.api.methods.JAXBProvider;
+import com.paloaltonetworks.panorama.api.methods.PanoramaApiClient;
+
+@Component(configurationPid = "com.paloaltonetworks.panorama.ApplianceManager",
+   property = { PLUGIN_NAME + "=Panorama", VENDOR_NAME + "=Palo Alto Networks", SERVICE_NAME + "=Panorama",
+                EXTERNAL_SERVICE_NAME + "=Pan-nsx", AUTHENTICATION_TYPE + "=BASIC_AUTH", NOTIFICATION_TYPE + "=NONE",
+                SYNC_SECURITY_GROUP + ":Boolean=true", PROVIDE_DEVICE_STATUS + ":Boolean=false",
+                SYNC_POLICY_MAPPING + ":Boolean=false", SUPPORT_MULTIPLE_POLICIES + ":Boolean=false" })
+public class PANApplianceManagerApi implements ApplianceManagerApi {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PANApplianceManagerApi.class);
+    private Client client;
+    private Config config;
+
+    @ObjectClassDefinition
+    @interface Config {
+        @AttributeDefinition(required = false)
+        boolean use_https() default true;
+
+        @AttributeDefinition(min = "0",
+                max = "65535",
+                required = false,
+                description = "The port to use when connecting to PAN instances. The value '0' indicates that a default port of '443' (or '80' if HTTPS is not enabled) should be used.")
+        int port() default 0;
+
+        @AttributeDefinition(min = "0", required = false)
+        int max_threads() default 0;
+
+        @AttributeDefinition(required = false,
+                description = "The property name to use when setting the maximum thread count")
+        String max_threads_property_name() default "com.sun.jersey.client.property.threadpoolSize";
+
+        @AttributeDefinition(required = false,
+                description = "PanOS id is a parameter required by commit operations")
+        String pan_os_id() default "007299000003740";
```

Comment:
`The target parameter on Panorama allows you to redirect queries to a managed firewall. Redirecting queries
to firewalls helps to reduce time and the number of steps required to issue repetitive commands. Using the
scripting language or your choice, you can store firewall serial numbers and use them to issue a query to
several firewalls`

this id is specific to a pan os device, so this will not work for different pan os and panorama instances....

did we try without a target parameter?